subsys/bluetooth/mesh/net.c

```diff
@@ -119,14 +119,20 @@ static bool check_dup(struct net_buf_simple *data)
 
 	val = sys_get_be32(tail - 4) ^ sys_get_be32(tail - 8);
 
-	for (i = 0; i < ARRAY_SIZE(dup_cache); i++) {
-		if (dup_cache[i] == val) {
+	for (i = dup_cache_next; i > 0;) {
+		if (dup_cache[--i] == val) {
+			return true;
+		}
+	}
+
+	for (i = ARRAY_SIZE(dup_cache); i > dup_cache_next;) {
+		if (dup_cache[--i] == val) {
```

Comment:
With this approach we will go through empty entries until `dup_cache_next` wraps around.

Reply:
The same is true for the original implementation.

Reply:
The original intention of this commit is to turn sequential iteration into reverse-order iteration.

Reply:
In fact, in a network, a cache will be filled soon, and I think it is meaningless to check whether it is blank.